src/Lachain.Core/RPC/HTTP/Web3/TransactionServiceWeb3.cs

```diff
@@ -568,9 +523,81 @@ public string GetNetworkGasPrice()
         [JsonRpcMethod("eth_signTransaction")]
         private string SignTransaction(JObject opts)
         {
-            // TODO: implement tx signing
-            return Web3DataFormatUtils.Web3Data("".HexToBytes());
-            //throw new ApplicationException("Not implemented yet");
+            var signedTx = MakeAndSignTransaction(opts);
+            var transaction = signedTx.Transaction;
+            var sign = signedTx.Signature;
+            var rawTx = transaction.RlpWithSignature(sign);
+            return Web3DataFormatUtils.Web3Data(rawTx);
+        }
+
+        private TransactionReceipt MakeAndSignTransaction(JObject opts){
+
+            var from = opts["from"];
+            var gas = opts["gas"];
+            var gasPrice = opts["gasPrice"];
+            var data = opts["data"];
+            var to = opts["to"];
+            var value = opts["value"];
+            var nonce = opts["nonce"];
+
+            if(from is null){
+                throw new ArgumentException("from should not be null");
+            }
+            var fromAddress = ((string) from!).HexToUInt160();
+
+
+            ulong? nonceToUse = _stateManager.LastApprovedSnapshot.Transactions.GetTotalTransactionCount(fromAddress);
+            if(!(nonce is null)) nonceToUse = ((string)nonce!).HexToUlong();
+
+            ulong? gasToUse = null;
+            if(!(gas is null)) gasToUse = ((string)gas!).HexToUlong();
+            
+            ulong? gasPriceToUse = null;
+            if(!(gasPrice is null)) gasPriceToUse  = ((string)gasPrice!).HexToUlong();
+            
+            byte[]? byteCode = null;
+            if(!(data is null)) byteCode = ((string) data!).HexToBytes();
+            
+            
+            var config = _configManager.GetConfig<VaultConfig>("vault") ??
+                         throw new Exception("No 'vault' section in config file");
+            string password = config.Password!;
```

Comment:
It is a wrong way. You can't access a config password here,  in this case anyone can send a transaction signed with this node and get access to any node money this way. Instead you should check if _privateWallet is unlocked and sign tx in this case only (throw an exception in other case). Also add this method to the private methods list,

Reply:
I added "eth_sendTransaction" and "eth_signTransaction" in private methods list.

Reply:
You still can't access config password in this case. Check _privateWallet is unlocked already during the call,  don't unlock it yourself

Reply:
done